api/accessManagement/index.ts

```diff
@@ -33,6 +36,84 @@ export async function refreshRoleAssignments(req, res: Response, next: NextFunct
   return res.send(refreshRoleAssignmentsSuccess);
 }
 
+// node layer logic for approving specific access request
+export async function approveSpecificAccessRequest(req, res: Response, next: NextFunction): Promise<Response> {
+  try {
+    // attempted to find specific requested role previously to get data but not neccessary
+    /* const caseRoles = await getRoleAssignmentForUser(req.body.person.id, req);
+    if (!caseRoles) {
+      return res.status(400);
+    }
+    req.body.restoreRole = caseRoles.find(role => role.id === req.body.requestId); */
+    // create the specific access approval role
+    const firstRoleResponse: AxiosResponse = await createSpecificAccessApprovalRole(req, res, next);
+    // 201
+    if (!firstRoleResponse || firstRoleResponse.status !== 201) {
+      return firstRoleResponse && firstRoleResponse.status
+       ? res.status(firstRoleResponse.status).send(firstRoleResponse) : res.status(400);
+    }
+    const deletionResponse = await deleteRoleByAssignmentId(req, res, next, req.body.requestId);
+    const rolesToDelete = firstRoleResponse.data.roleAssignmentResponse.requestedRoles;
+    if (!deletionResponse || deletionResponse.status !== 204) {
+      // delete the roles created previously
+      return deleteSpecificAccessRoles(req, res, next, deletionResponse, rolesToDelete);
+    }
+    // 204
+    const taskResponse: AxiosResponse = await postTaskCompletionForAccess(req, res, next);
+    if (!taskResponse || taskResponse.status !== 204) {
+      // restore specific access requested role and delete two created roles
+      return restoreDeletedRole(req, res, next, taskResponse, rolesToDelete);
+    }
+    // if everything has worked send the last response back to the user
+    return res.send(taskResponse.data).status(taskResponse.status);
+  } catch (error) {
+    next(error);
+    return res.status(error.status).send(error);
+  }
+}
+
+// attempts to delete
+// tslint:disable-next-line:max-line-length
+export async function deleteSpecificAccessRoles(req, res: Response, next: NextFunction, previousResponse: AxiosResponse<any>, rolesToDelete: any): Promise<Response> {
```

Comment:
`rolesToDelete`  Can this be strongly typed?

Reply:
yes, I'll do that